### PR TITLE
🌱 Fix FROM_K8S_VERSION variable v1.30.0 -> v1.31.2

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Fix FROM_K8S_VERSION variable v1.30.0 -> v1.31.2

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by adilGhaffarDev** [comment]
/hold
I will unhold  this after the release 1.9.0

**Comment by adilGhaffarDev** [comment]
seems like a flake, i will try one more time
/test metal3-centos-e2e-feature-test-main-pivoting

**Review by tuminoid — COMMENTED** [review]
/approve

File: test/e2e/upgrade_clusterctl_test.go [hunk 1]
```diff
@@ -109,8 +109,8 @@ var _ = Describe("When testing cluster upgrade from releases (v1.7=>current) [cl
 			InitWithBootstrapProviders:      []string{fmt.Sprintf(providerKubeadmPrefix, capiStableRelease)},
 			InitWithControlPlaneProviders:   []string{fmt.Sprintf(providerKubeadmPrefix, capiStableRelease)},
 			InitWithInfrastructureProviders: []string{fmt.Sprintf(providerMetal3Prefix, capm3StableRelease)},
-			InitWithKubernetesVersion:       "v1.30.0",
-			WorkloadKubernetesVersion:       "v1.30.0",
+			InitWithKubernetesVersion:       "v1.31.2",
+			WorkloadKubernetesVersion:       "v1.31.2",
```
Comment (kashifest) [inline test/e2e/upgrade_clusterctl_test.go:113]: why are we changing this for `minorVersion := "1.7"`?

Comment (adilGhaffarDev) [reply test/e2e/upgrade_clusterctl_test.go:113]: upgrade from v1.30.0 to v1.32.0 is not going to work. This test needs to be updated. I will remove this and open separate PR to update clusterctl tests for the main branch.

Comment (kashifest) [reply test/e2e/upgrade_clusterctl_test.go:113]: sorry got confused, remove what?

Comment (adilGhaffarDev) [reply test/e2e/upgrade_clusterctl_test.go:113]: yes this is not needed I am removing it.

**Review by kashifest — COMMENTED** [review]
/lgtm

File: test/e2e/upgrade_clusterctl_test.go (hunk 1 above)
Comment (adilGhaffarDev) [reply test/e2e/upgrade_clusterctl_test.go]: my bad I thought this test was updating k8s too, but clusterctl tests only upgrade providers, not k8s. So this change is not needed, I have removed it.

**Pull request merged**